apps/parsley/src/utils/filterLogs/index.ts

```diff
@@ -80,20 +80,26 @@ const filterLogs = (options: FilterLogsParams): ProcessedLogLines => {
       for (let idx = 0; idx < logLines.length; idx++) {
         const func = sectionData.functions[funcIndex];
         const isFuncStart = func && idx === func.range.start;
-        const isFuncOpen = sectionState[func?.functionID]?.isOpen ?? false;
-
+        const isFuncOpen = sectionState[func?.functionID]?.isOpen;
+        const renderCommands = isFuncOpen || func?.containsTopLevelCommand;
         const command = sectionData.commands[commandIndex];
         const isCommandStart = command && idx === command.range.start;
 
+        // A function start is detected.
         if (isFuncStart) {
-          // A function start is detected.
-          filteredLines.push({
-            ...func,
-            isOpen: isFuncOpen,
-            rowType: RowType.SectionHeader,
-          });
+          // Don't show the function header if contains a top-level command.
+          if (!func.containsTopLevelCommand) {
+            const { functionID, functionName, range } = func;
+            filteredLines.push({
+              functionID,
+              functionName,
+              isOpen: isFuncOpen,
+              range,
+              rowType: RowType.SectionHeader,
+            });
+          }
           funcIndex += 1;
-          if (!isFuncOpen) {
+          if (!renderCommands) {
             // The function is closed. Skip all log lines until the end of the function.
```

Comment:
I'm not sure if this comment is clear after the updates (i.e. why does `!renderCommands` mean the function is closed?)

Reply:
Good point, I updated the comment.